File: server/src/models/Rating.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/db');

const Rating = sequelize.define('Rating', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  sessionId: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  matchId: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  raterId: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  rateeId: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  role: {
    type: DataTypes.ENUM('teacher', 'learner'),
    allowNull: false
  },
  stars: {
    type: DataTypes.INTEGER,
    allowNull: false,
    validate: { min: 1, max: 5 }
  },
  feedback: {
    type: DataTypes.TEXT
  }
}, {
  tableName: 'ratings',
  timestamps: true
});

module.exports = Rating;

File: server/src/main/java/com/skillswap/entity/Rating.java
package com.skillswap.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;

@Entity
@Table(name = "ratings")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Rating {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long sessionId;

    @Column(nullable = false)
    private Long matchId;

    @Column(nullable = false)
    private Long raterId;

    @Column(nullable = false)
    private Long rateeId;

    @Column(nullable = false, length = 10)
    private String role; // teacher, learner

    @Column(nullable = false)
    private Integer stars;

    @Column(columnDefinition = "TEXT")
    private String feedback;

    @CreationTimestamp
    private LocalDateTime createdAt;
}
